provider/core/src/response.rs

```diff
@@ -74,38 +69,68 @@ pub struct DataResponseMetadata {
 pub struct DataPayload<M: DataMarker>(pub(crate) DataPayloadInner<M>);
 
 pub(crate) enum DataPayloadInner<M: DataMarker> {
-    Yoke(Yoke<M::Yokeable, Option<Cart>>),
+    Yoke(Yoke<M::Yokeable, Cart>),
     StaticRef(&'static M::Yokeable),
 }
 
 /// The type of the "cart" that is used by `DataPayload`.
 #[derive(Clone, Debug)]
 #[allow(clippy::redundant_allocation)] // false positive, it's cheaper to wrap an existing Box in an Rc than to reallocate a huge Rc
-pub struct Cart(SelectedRc<Box<[u8]>>);
+pub struct Cart(OptionSelectedRcBytes);
```

Comment:
I think it would be nicer if you made this `OptionSelectedRc<Box<[u8]>>`. I don't think anything in `OptionSelectedRcBytes` cannot be made generic, and it's an easier type if it's generic.

Reply:
I made it generic in https://github.com/unicode-org/icu4x/pull/4449

Reply:
I disagree that it's easier if generic, there were a bunch of subtle safety concerns that we were able to skip by making it not-generic